parsimonious/expressions.py

```diff
@@ -324,18 +329,19 @@ def resolve_refs(self, rule_map):
         self.members = tuple(m.resolve_refs(rule_map) for m in self.members)
         return self
 
+    def _eq_no_recursion(self, other, checked):
```

Comment:
The name doesn't make sense outside the context of an infinite recursion error being on your mind. `_eq_no_recursion`recursively calls `_eq_no_recursion`? 

Something like like `_eq_check_cycles` or even just `_eq`.

Reply:
true... I wanted to do it non recursively at first but forgot it :laughing: thanks for your comment!